src/typing_extensions.py

```diff
@@ -2809,6 +2811,44 @@ def get_protocol_members(tp: type, /) -> typing.FrozenSet[str]:
             return frozenset(tp.__protocol_attrs__)
         return frozenset(_get_protocol_attrs(tp))
 
+if hasattr(typing, "doc"):
+    doc = typing.doc
+    DocInfo = typing.DocInfo
+else:
+    class DocInfo:
+        """Container for documentation information as returned by ``doc()``.
+
+        It is expected this class wouldn't be manually created but instead returned
+        by ``doc()`` inside of type annotations using ``Annotated``.
+
+        The ``documentation`` attribute contains the documentation string passed
+        to ``doc()``.
+        """
+        def __init__(self, documentation: str):
+            self.documentation = documentation
+
+        def __repr__(self) -> str:
+            return f"DocInfo({self.documentation!r})"
```

Comment:
```suggestion
            return f"DocInfo({self.documentation!r})"
       
       def __hash__(self) -> int:
            return hash(self.documentation)
```
Let's make it hashable (hope I got the indentation right)